server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -160,17 +165,26 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
+
         try {
             // Will be replaced with ExtensionTokenProcessor and PrincipalIdentifierToken classes from feature/identity
             final String extensionTokenProcessor = "placeholder_token_processor";
             final String requestIssuerIdentity = "placeholder_request_issuer_identity";
 
+            this.allowList = List.of("Content-Type");
+            this.denyList = List.of("Authorization", "Proxy-Authorization");
+
+            Map<String, List<String>> filteredHeaders = headers.entrySet()
```

Comment:
When making this set of headers, there is a check for `allowList.contains("*")` that doesn't seem usable - it should be removed to keep the code cleaner.